samay.js

```diff
@@ -2,6 +2,32 @@
 
 function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
 
+/* global window */
+
+function calcDate(structuredDate) {
+
+	try {
+		var splitDate = structuredDate.split('T'),
+		    datePart = splitDate[0].split('-'),
```

Comment:
Might want to check if the date exists?

Reply:
Enclosed in try catch. As per @vihangpatel